src/openloco/localisation/languagefiles.cpp

```diff
@@ -0,0 +1,274 @@
+                }
+                else if (commands[0] == "RAWDATE" && commands.size() >= 2)
+                {
+                    *out++ = (char)control_codes::date;
+                    if (commands.size() == 3 && commands[1] == "MY" && commands[2] == "SHORT")
+                    {
+                        *out++ = date_modifier::raw_my_abbr;
+                    }
+                }
+                else if (commands[0] == "DATE" && commands.size() == 2)
+                {
+                    *out++ = (char)control_codes::date;
+                    if (commands[1] == "DMY")
+                    {
+                        *out++ = date_modifier::dmy_full;
+                    }
+                    else if (commands[1] == "MY")
+                    {
+                        *out++ = date_modifier::my_full;
+                    }
+                }
+                else if (commands[0] == "COLOUR")
+                {
+                    auto colour = textColourNames.find(commands[1]);
+                    if (colour != textColourNames.end())
+                    {
+                        *out = colour->second;
+                        out++;
+                    }
+                    else
+                    {
+                        printf("%.*s\n", (int)commands[1].length(), commands[1].data());
+                    }
+                }
+                else if (commands[0] == "MOVE_X")
+                {
+                    *out++ = (char)control_codes::move_x;
+                    uint8_t pixels_to_move_by = std::atoi(commands[1].data());
+                    *out++ = pixels_to_move_by;
+                }
+                else if (commands[0] == "NEWLINE")
+                {
+                    if (commands.size() == 1)
+                    {
+                        *out++ = (char)control_codes::newline;
+                    }
+                    else if (commands.size() == 3)
+                    {
+                        *out++ = (char)control_codes::newline_x_y;
+
+                        uint8_t x_pixels_to_move_by = std::atoi(commands[1].data());
+                        *out++ = x_pixels_to_move_by;
+
+                        uint8_t y_pixels_to_move_by = std::atoi(commands[2].data());
+                        *out++ = y_pixels_to_move_by;
+                    }
+                }
+                else
+                {
+                    printf("%.*s\n", (int)commands[0].length(), commands[0].data());
+                }
+
+                continue;
+            }
+            else
+            {
+                *out = readChar;
+                out++;
+            }
+
+            if (readChar == '\0')
+                break;
+        }
+
+        return str;
+    }
+
+    static bool stringIsBuffer(int id)
+    {
+        return id == string_ids::buffer_337 || id == string_ids::buffer_338 || id == string_ids::buffer_1250 || id == string_ids::preferred_currency_buffer || id == string_ids::buffer_1719
```

Comment:
maybe put each case on a separate line to make for easier refactoring

Reply:
`clang-format` reverts it to this form. I don't mind disabling it for this expression, but don't feel strongly about this either.

Reply:
It depends on where you place the `||`, iirc